src/node_snapshotable.cc

```diff
@@ -1168,31 +1159,32 @@ StartupData SerializeNodeContextInternalFields(Local<Object> holder,
                      static_cast<int>(info->length)};
 }
 
-void SerializeBindingData(Environment* env,
-                          SnapshotCreator* creator,
-                          EnvSerializeInfo* info) {
+void SerializeSnapshotableObjects(Environment* env,
+                                  SnapshotCreator* creator,
+                                  EnvSerializeInfo* info) {
   uint32_t i = 0;
-  env->ForEachBindingData([&](FastStringKey key,
-                              BaseObjectPtr<BaseObject> binding) {
+  env->ForEachBaseObject([&](BaseObject* obj) {
+    if (!obj->is_snapshotable()) {
+      return;
+    }
+    SnapshotableObject* ptr = static_cast<SnapshotableObject*>(obj);
+
+    const char* type_name = ptr->GetTypeNameChars();
     per_process::Debug(DebugCategory::MKSNAPSHOT,
-                       "Serialize binding %i (%p), object=%p, type=%s\n",
+                       "Serialize snapshotable object %i (%p), "
+                       "object=%p, type=%s\n",
                        static_cast<int>(i),
-                       binding.get(),
-                       *(binding->object()),
-                       key.c_str());
+                       ptr,
+                       *(ptr->object()),
+                       type_name);
 
-    if (IsSnapshotableType(key)) {
-      SnapshotIndex index = creator->AddData(env->context(), binding->object());
+    if (ptr->PrepareForSerialization(env->context(), creator)) {
+      SnapshotIndex index = creator->AddData(env->context(), obj->object());
       per_process::Debug(DebugCategory::MKSNAPSHOT,
                          "Serialized with index=%d\n",
                          static_cast<int>(index));
-      info->bindings.push_back({key.c_str(), i, index});
-      SnapshotableObject* ptr = static_cast<SnapshotableObject*>(binding.get());
-      ptr->PrepareForSerialization(env->context(), creator);
-    } else {
-      UNREACHABLE();
```

Comment:
I assume that the removal of UNREACHABLE here and skipping preparation for serialization is delaying the check to v8's unserialized global handles check, as it prints more detailed debug information for these objects. Would it be worth documenting here about it?

Reply:
What should we document? `If there is a BaseObject that is not Snapshotable in the snapshot, V8 would crash and print more information about it`?

Reply:
When reading the code, I was thinking about what would happen if the non-snapshotable objects are skipped here. I have to compile it and play around with it until I realize v8 is going to reject creating the snapshot if any global handle is not serialized. So I'm thinking of a note about "The non-snapshotable objects are printed for debugging and fail the snapshot building later" can be helpful.

Reply:
Added some comments in the `!obj->is_snapshotable()` branch above